client/sysinfo/src/sysinfo.rs

```diff
@@ -32,6 +33,77 @@ use std::{
 	time::{Duration, Instant},
 };
 
+/// Throughput as measured in bytes per second.
+#[derive(Deserialize, Serialize, Debug, Clone, Copy, PartialEq)]
+pub enum Throughput {
+	/// KiB/s
+	KiBs(f64),
+	/// MiB/s
+	MiBs(f64),
+	/// GiB/s
+	GiBs(f64),
+}
```

Comment:
Hm... now that I look at it I find having this as an enum a little strange; can we refactor this to be a normal newtype?

Simply put, something like this:

```
pub struct Throughput(f64);
```

This would keep the results at full bytes/s precision.

Also, please note that either way this requires either a) a custom implementation of `Serialize`, or annotating the `HwBench` structure with a `#[serde(serialize_with = "...")]` to preserve the format which the telemetry expects (which is always in MB/s). Since this is going to hold the full precision of values yet the serialization is only in MB/s it feels more natural to me to go with the option B.

@ggwpez Thoughts?

Reply:
That is a very good idea! And then the `Display` implementation could show the best fitting formatting depending on the size of the value.

Reply:
Yes.

Reply:
I chose that enum originally to make use of serde to have nice JSON formats eg:  
```json
"MiBs": 450.0
```
But we can probably re-build that through the serde deserialization, or?

Reply:
> But we can probably re-build that through the serde deserialization, or?

Yes. For where we need a pretty output for human consumable JSON we can just easily attach a custom serializer, so that shouldn't be a problem.

Reply:
We could also consider moving the `Metric` type, but that would only be for uniform naming.